### PR TITLE
Remove Kotlin from third party libs.

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Refactoring

### Description

This PR removes the dependency on OkHttp and thus Kotlin in a fully backwards compatible fashion from the third-party libs for Kafka 3.6 and 3.7. It does this by replacing the opentelemetry-exporter-sender-okhttp (which is the OpenTelemetry default) with two separate senders. One for each transport.

- `opentelemetry-exporter-sender-jdk` enables sending traces using http/protobuf which is the lowest common denominator option, as its the only method required by the OpenTelemetry specifications.
- `opentelemetry-exporter-sender-grpc-managed-channel` enables support for sending traces using grpc and thus feature parity with the OkHttp sender.

This is the same approach as taken in https://github.com/strimzi/strimzi-kafka-bridge/pull/867

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by ppatierno** [comment]
@SamBarker AFAICS, the pipeline is failing because you didn't update the version for the `opentelemetry-sdk-extension-autoconfigure` artifact. It's still on the "old" `opentelemetry.alpha-version` property while it should use the ne w `opentelemetry.version` property as you did for the bridge.

**Comment by SamBarker** [comment]
I think the PR is ready for review the build failure appears un-related but I'll try and re run locally to validate.

```
[ERROR] Failures: 
[ERROR] io.strimzi.operator.cluster.operator.assembly.KafkaConnectApiIT.testChangeLoggers(VertxTestContext)
[ERROR]   Run 1: KafkaConnectApiIT.testChangeLoggers(VertxTestContext) java.lang.AssertionError: 
Expected: is "INFO"
     but: was null
[ERROR]   Run 2: KafkaConnectApiIT.testChangeLoggers(VertxTestContext) org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
[ERROR]   Run 3: KafkaConnectApiIT.testChangeLoggers(VertxTestContext) org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
[INFO] 
[INFO] 
[ERROR] Tests run: 18, Failures: 1, Errors: 0, Skipped: 0
```

I don't have enough permissions to assign a milestone

**Comment by ppatierno** [comment]
@SamBarker this problem should have been fixed and it's in the main branch now. Can you try to rebate your branch and see if the failures go away? Thanks.

**Comment by SamBarker** [comment]
> @SamBarker this problem should have been fixed and it's in the main branch now. Can you try to rebate your branch and see if the failures go away? Thanks.

Awesome, rebasing has resolved that problem. All the PR build jobs are passing.

**Comment by ppatierno** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --profile=all --groups=tracing

**Comment by ppatierno** [comment]
@SamBarker FYI I had to push a couple of missing changes to the PR which were causing the OpenTelemetry regression tests failing with the current exception ... 

```shell
2024-03-05 12:54:37,791 ERROR [my-cluster-0b07b447|task-0] Failed to start task my-cluster-0b07b447-0 (org.apache.kafka.connect.runtime.Worker) [StartAndStopExecutor-connect-my-cluster-0b07b447-connect-0.my-cluster-0b07b447-connect.namespace-19.svc:8083-2]
org.apache.kafka.common.KafkaException: Failed to construct kafka consumer
	at org.apache.kafka.clients.consumer.internals.LegacyKafkaConsumer.<init>(LegacyKafkaConsumer.java:264)
	at org.apache.kafka.clients.consumer.internals.ConsumerDelegateCreator.create(ConsumerDelegateCreator.java:65)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:600)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:595)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:543)
	at org.apache.kafka.connect.runtime.Worker$SinkTaskBuilder.doBuild(Worker.java:1807)
	at org.apache.kafka.connect.runtime.Worker$TaskBuilder.build(Worker.java:1754)
	at org.apache.kafka.connect.runtime.Worker.startTask(Worker.java:664)
	at org.apache.kafka.connect.runtime.Worker.startSinkTask(Worker.java:537)
	at org.apache.kafka.connect.runtime.distributed.DistributedHerder.startTask(DistributedHerder.java:1968)
	at org.apache.kafka.connect.runtime.distributed.DistributedHerder.lambda$getTaskStartingCallable$38(DistributedHerder.java:2018)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.apache.kafka.common.KafkaException: Class io.opentelemetry.instrumentation.kafkaclients.TracingConsumerInterceptor cannot be found
	at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstance(AbstractConfig.java:392)
	at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstances(AbstractConfig.java:483)
	at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstances(AbstractConfig.java:461)
	at org.apache.kafka.clients.ClientUtils.configuredInterceptors(ClientUtils.java:260)
	at org.apache.kafka.clients.consumer.internals.ConsumerUtils.configuredConsumerInterceptors(ConsumerUtils.java:160)
	at org.apache.kafka.clients.consumer.internals.LegacyKafkaConsumer.<init>(LegacyKafkaConsumer.java:174)
	... 14 more
Caused by: java.lang.ClassNotFoundException: io.opentelemetry.instrumentation.kafkaclients.TracingConsumerInterceptor
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
	at org.apache.kafka.connect.runtime.isolation.PluginClassLoader.loadClass(PluginClassLoader.java:124)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at org.apache.kafka.common.utils.Utils.loadClass(Utils.java:426)
	at org.apache.kafka.common.utils.Utils.newInstance(Utils.java:415)
	at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstance(AbstractConfig.java:390)
	... 19 more`
```

The Kafka clients instrumentation has the interceptors in a different package in version 1.32.0-alpha (actually they added a subpackage `v2_6`) but the operator was still configuring the old one.

**Comment by SamBarker** [comment]
>@SamBarker FYI I had to push a couple of missing changes to the PR which were causing the OpenTelemetry regression tests failing with the current exception ...

urgh a bit sad that it needs to get all the way to a system tests to catch that. I'll try again to get them running locally. Thanks @ppatierno

File: tracing-agent/src/main/java/io/strimzi/tracing/agent/OpenTelemetryTracing.java [hunk 1]
```diff
@@ -20,7 +20,8 @@ public void initialize() {
         String serviceName = System.getenv("OTEL_SERVICE_NAME");
         if (serviceName != null) {
             LOGGER.info("Initializing OpenTelemetry tracing with service name {}", serviceName);
-            System.setProperty("otel.metrics.exporter", "none"); // disable metrics
+            System.setProperty("otel.metrics.exporter", "none"); // disable otel metrics exporter. This has no effect on metrics in the bridge.
+            System.setProperty("otel.logs.exporter", "none"); // disable otel logs exporter. This has no effect on logging in the bridge.
```
Comment (SamBarker) [inline tracing-agent/src/main/java/io/strimzi/tracing/agent/OpenTelemetryTracing.java:24]: @ppatierno I'm not sure what context the tracing agent is used, but its certainly not `in the bridge` 😜 How about
```suggestion
            System.setProperty("otel.metrics.exporter", "none"); // disable otel metrics exporter. This has no effect on metrics in Strimzi.
            System.setProperty("otel.logs.exporter", "none"); // disable otel logs exporter. This has no effect on logging in Strimzi.
```

Comment (ppatierno) [reply tracing-agent/src/main/java/io/strimzi/tracing/agent/OpenTelemetryTracing.java:24]: Yeah sure. Blind copy/paste grrrrrrrr
Please update this way.

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,6 +50,14 @@
...
   We expect this to improve the compatibility with various tools used for scaling Kubernetes clusters such as [Karpenter](https://karpenter.sh/).
   If you observe any problems with your toolchain or just want to stick with the previous behavior, you can use the `STRIMZI_DENY_EVICTION` environment variable and set it to `false` to switch back to the old (legacy) mode.
+* The following dependencies have been removed from the third-party libs.
+  * `com.squareup.okhttp3:okhttp:jar:4.10.0`
+  * `com.squareup.okio:okio-jvm:jar:3.0.0`
+  * `org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.5.31`
+  * `org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.5.31`
+  * `org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.5.31`
+  * `org.jetbrains.kotlin:kotlin-stdlib:jar:1.6.20`
+  * `org.jetbrains:annotations:jar:13.0`
```
Comment (scholzj) [inline CHANGELOG.md:60]: This has no meaning for the user. Please remove it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## 0.40.0
 
+* Migrate from OpenTelemetry tracing from using OkHttp sender to the JDK sender or GRPC sender thus remove the kotlin dependency.
```
Comment (scholzj) [inline CHANGELOG.md:5]: This has no meaning for the users. If it has any user impact, please explain it. If not, please remove it.

File: CHANGELOG.md (hunk 2 above)
Comment (ppatierno) [reply CHANGELOG.md:60]: Yeah, I was going to make the same comment. These are details not really useful to the user imho. Same for the next comment.

File: CHANGELOG.md (hunk 3 above)
Comment (SamBarker) [reply CHANGELOG.md]: Nope there is no user impact, I have removed.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**